samples/Calling/src/app/utils/ShakeToSendLogs.tsx

```diff
@@ -0,0 +1,238 @@
+    subText:
+      'Would you like to enable shake to help? This will prompt you to send device logs when you shake your device.'
+  };
+
+  return (
+    <Dialog hidden={!showPrompt} dialogContentProps={dialogContentProps}>
+      <DialogFooter>
+        <PrimaryButton
+          text="Enable"
+          onClick={() => {
+            requestPermission().then((res) => {
+              if (res === 'granted') {
+                props.onPermissionGranted();
+              }
+              setShowPrompt(false);
+            });
+          }}
+        />
+        <DefaultButton onClick={closePrompt} text="Disable" />
+      </DialogFooter>
+    </Dialog>
+  );
+};
+
+export const ShakeToSendLogs = (): JSX.Element => {
+  const needsToRequestDeviceMotionPermission = !!(DeviceOrientationEvent as unknown as DeviceMotionEventiOS)
+    ?.requestPermission;
+  const [hasPermission, setHasPermission] = React.useState(!needsToRequestDeviceMotionPermission);
+  const [showDialog, closeDialog] = useShakeDialog(hasPermission);
+
+  const [logStatus, setLogStatus] = React.useState<'unsent' | 'sending' | 'failed' | 'sent'>('unsent');
+  const [blobUrl, setBlobUrl] = React.useState<string>();
+
+  const reset = (): void => {
+    setLogStatus('unsent');
+    setBlobUrl(undefined);
+  };
+
+  const onSendLogsClick = async (): Promise<void> => {
+    setLogStatus('sending');
+    try {
+      const result = await sendLogs();
+      if (result) {
+        setLogStatus('sent');
+        setBlobUrl(result);
+      } else {
+        setLogStatus('failed');
+      }
+    } catch {
+      setLogStatus('failed');
+    }
+  };
+
+  const dialogContentProps = {
+    type: DialogType.normal,
+    title: logStatus === 'sent' ? 'Logs sent!' : 'Send logs',
+    subText: logStatus === 'sent' ? undefined : 'We detected a shake. Would you like to send logs to help us debug?'
+  };
+
+  return (
+    <>
+      {needsToRequestDeviceMotionPermission && (
+        <PromptForShakePermission onPermissionGranted={() => setHasPermission(true)} />
+      )}
+      <Dialog hidden={!showDialog} dialogContentProps={dialogContentProps} modalProps={{ onDismissed: reset }}>
+        <Spinner hidden={logStatus !== 'sending'} label="Sending logs..." />
+        {logStatus === 'unsent' && (
+          <DialogFooter>
+            <PrimaryButton onClick={onSendLogsClick} text="Send" />
+            <DefaultButton onClick={closeDialog} text="Don't send" />
+          </DialogFooter>
+        )}
+        {logStatus === 'sent' && (
+          <>
+            <Text>
+              Your logs were successfully uploaded. You can access them{' '}
+              <Link href={blobUrl} target="_blank">
+                here
+              </Link>
+              .
+            </Text>
+            <DialogFooter>
+              <PrimaryButton
+                text="Send as email"
+                onClick={() => window.open(`mailto:?body=${encodeURIComponent(blobUrl ?? '')}`)}
```

Comment:
is this email something that we provide at send time? or something we setup into the server?

Reply:
its a standard protocol that just opens your default mail client with the body attached! So doesn't send the email itself just opens a client for you to send it